core/templates/components/summary-tile/story-summary-tile.directive.ts

```diff
@@ -16,34 +16,61 @@
  * @fileoverview Component for a canonical story tile.
  */
 
+require(
+  'components/summary-tile/' +
+  'story-summary-tile-displayed-chapters-calculator.directive.ts');
+
 require('domain/utilities/url-interpolation.service.ts');
 require('domain/topic_viewer/topic-viewer-domain.constants.ajs.ts');
+require('services/assets-backend-api.service.ts');
+require('services/contextual/window-dimensions.service.ts');
 
 angular.module('oppia').directive('storySummaryTile', [
   'UrlInterpolationService', function(UrlInterpolationService) {
     return {
       restrict: 'E',
       scope: {},
       bindToController: {
-        getStoryId: '&storyId',
-        getStoryTitle: '&title',
-        getStoryDescription: '&description',
+        getStorySummary: '&storySummary'
       },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/components/summary-tile/story-summary-tile.directive.html'),
       controllerAs: '$ctrl',
-      controller: ['STORY_VIEWER_URL_TEMPLATE',
-        function(STORY_VIEWER_URL_TEMPLATE) {
+      controller: [
+        'AssetsBackendApiService', 'WindowDimensionsService',
+        'ENTITY_TYPE', 'STORY_VIEWER_URL_TEMPLATE',
+        function(
+            AssetsBackendApiService, WindowDimensionsService,
+            ENTITY_TYPE, STORY_VIEWER_URL_TEMPLATE) {
           var ctrl = this;
           ctrl.getStoryLink = function() {
             return UrlInterpolationService.interpolateUrl(
               STORY_VIEWER_URL_TEMPLATE, {
-                story_id: ctrl.getStoryId()
+                story_id: ctrl.getStorySummary().getId()
               });
           };
 
-          ctrl.getStaticImageUrl = function(imagePath) {
-            return UrlInterpolationService.getStaticImageUrl(imagePath);
+          ctrl.showAllChapters = function() {
+            ctrl.stopWatcherOnHeight = true;
```

Comment:
Not really sure, from the varname, what this means.

Reply:
Oh, this is an unused variable, will remove it.

Reply:
Done